internal/vulnstore/jsonblob/jsonblob.go

```diff
@@ -0,0 +1,199 @@
+package jsonblob
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/libvuln/driver"
+)
+
+// New constructs an empty Store.
+func New() (*Store, error) {
+	s := Store{}
+	s.ops = make(map[string][]driver.UpdateOperation)
+	s.entry = make(map[uuid.UUID]*Entry)
+	return &s, nil
+}
+
+// A Store buffers update operations.
+type Store struct {
+	sync.RWMutex
+	entry  map[uuid.UUID]*Entry
+	ops    map[string][]driver.UpdateOperation
+	latest uuid.UUID
+}
+
+// Load reads in all the records serialized in the provided Reader.
+func Load(ctx context.Context, r io.Reader) <-chan *Entry {
+	// This is a crappy API, but doing it this way limit memory usage.
+	ch := make(chan *Entry)
+
+	go func() {
+		defer close(ch)
+		var err error
+		var de diskEntry
+		cur := uuid.Nil
+		var e *Entry
+		dec := json.NewDecoder(r)
+		for err = dec.Decode(&de); err == nil; err = dec.Decode(&de) {
```

Comment:
just a quick comment on the logic in this loop would go a long way to quickly understanding the channel usage, packing of entry and concerns over memory here.